substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java

```diff
@@ -384,6 +405,37 @@ private static void linkSunEC(DuringAnalysisAccess a) {
         }
     }
 
+    private static void linkSunMSCAPI(DuringAnalysisAccess a) {
+        NativeLibraries nativeLibraries = ((FeatureImpl.DuringAnalysisAccessImpl) a).getNativeLibraries();
+        /* We statically link sunmscapi thus we classify it as builtIn library */
+        NativeLibrarySupport.singleton().preregisterUninitializedBuiltinLibrary("sunmscapi");
+
+        nativeLibraries.addStaticJniLibrary("sunmscapi");
+        /* Library sunmscapi depends on ncrypt and crypt32 */
+        nativeLibraries.addDynamicNonJniLibrary("ncrypt");
+        nativeLibraries.addDynamicNonJniLibrary("crypt32");
+    }
+
+    private static void registerLoadKeysOrCertificateChains(DuringAnalysisAccess a) {
+        JNIRuntimeAccess.register(clazz(a, "java.util.ArrayList"));
```

Comment:
Not necessary, because the constructor/method registration will also register the class.
```suggestion
```

Reply:
done